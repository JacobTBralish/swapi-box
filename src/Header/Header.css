li, a {
  list-style: none;
}

a:visited, a {
  color: black;
  text-decoration: none;
}