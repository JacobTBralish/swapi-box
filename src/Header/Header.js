import React from 'react';
import { NavLink } from 'react-router-dom';
import './Header.css'

const Header = (props) => {
  return(
    <header>
      <h1><NavLink to='/'>SWAPI-BOX</NavLink></h1>
      <nav>
        <ul>
          <li>
            <NavLink to='/people'>People</NavLink>
          </li>
          <li>
            <NavLink to='/planets'>Planets</NavLink>
          </li>
          <li>
            <NavLink to='/vehicles'>Vehicles</NavLink>
          </li>
          <li>
            <NavLink to='/favorites'>Favorites</NavLink>
          </li>
        </ul>
      </nav>
    </header>
  )
}

export default Header;