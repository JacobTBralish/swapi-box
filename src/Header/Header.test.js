import React from 'react';
import ReactDOM from 'react-dom';
import Header from './Header';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

describe('Header Component', () => {
  test('that it should mount with crashing', () => {
    const renderedComp = shallow(<Header />)

    expect(renderedComp).toMatchSnapshot()
  })

  test('that it should contain a navLink to the home page', () => {
    const renderedComp = shallow(<Header />)

    expect(renderedComp.find('NavLink').first().props().to).toEqual('/')
  })

  test('that it should contain 5 NavLinks', () => {
    const renderedComp = shallow(<Header />)

    expect(renderedComp.find('NavLink').length).toEqual(5)
  })
})