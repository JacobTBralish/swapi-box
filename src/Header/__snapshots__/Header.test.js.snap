// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component that it should mount with crashing 1`] = `
<header>
  <NavLink
    activeClassName="active"
    ariaCurrent="true"
    className="logo"
    to="/"
  >
    <img
      alt="star-wars-logo"
      src="star-wars-logo.png"
    />
    <h1
      className="home-header"
    >
      THE API STRIKES BACK
    </h1>
  </NavLink>
  <nav>
    <ul>
      <li
        className="nav-item"
      >
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          to="/people"
        >
          People
        </NavLink>
      </li>
      <li
        className="nav-item"
      >
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          to="/planets"
        >
          Planets
        </NavLink>
      </li>
      <li
        className="nav-item"
      >
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          to="/vehicles"
        >
          Vehicles
        </NavLink>
      </li>
      <li
        className="nav-item"
      >
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          to="/favorites"
        >
          Favorites
        </NavLink>
      </li>
    </ul>
  </nav>
</header>
`;
